slsim/image_simulation.py

```diff
@@ -395,10 +417,180 @@ def point_source_image_with_variability(
             mag_zero_point=mag_zero,
             delta_pix=delta_pix,
             num_pix=num_pix,
-            psf_kernels=psf_kernel,
+            psf_kernel=psf_kernel,
             transform_pix2angle=transf_matrix,
             time=time,
         )
         all_image.append(image_test)
     variab_images = [list(x) for x in zip(*all_image)]
-    return variab_images
+    image = np.array(variab_images)
+    n = len(image)
+    subarray_shape = image[0].shape
+    results = np.zeros(subarray_shape)
+    for i in range(n):
+        results += image[i]
+    return results
+
+
+def deflector_images_with_different_zeropoint(
+    lens_class, band, mag_zero_point, delta_pix, num_pix
+):
+    """Creates deflector images with different magnitude zero point.
+
+    :param lens_class: Lens() object
+    :param band: imaging band
+    :param mag_zero_point: magnitude zero point in band
```

Comment:
you should specify that this is a list. the entire point of this function. Also describe this function in more detail